app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :likes
-  has_many :comments
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  after_create :update_counter
+
+  def update_counter
+    user.update(posts_counter: user.posts.size)
+  end
+
+  def recent_comments
+    comments.last(5)
+  end
```

Comment:
- [ ] I suggest that comments be DESC order as well like `comments.order('created_at Desc').limit(5)`